File: ASProj/hi-ui/src/main/java/com/example/hiui/refresh/HiRefresh.java
package com.example.hiui.refresh;

public interface HiRefresh {
    /**
     * 刷新时是否禁止滚动
     * @param disableRefreshScroll
     */
    void setDisableRefreshScroll(boolean disableRefreshScroll);

    /**
     * 刷新完成
     */
    void refreshFinished();

    void setRefreshOverView(HiOverView hiOverView);
    /**
     * 设置下拉刷新的监听器
     * @param hiRefreshListener
     */
    void setRefreshListener(HiRefresh.HiRefreshListener hiRefreshListener);

    interface HiRefreshListener{
        void onRefresh();

        boolean enableRefresh();
    }
}

File: ASProj/hi-ui/src/main/java/com/example/hiui/refresh/HiRefresh.kt
package com.example.hiui.refresh

interface HiRefresh {
    /**
     * 刷新时是否禁止滚动
     * @param disableRefreshScroll
     */
    fun setDisableRefreshScroll(disableRefreshScroll: Boolean)

    /**
     * 刷新完成
     */
    fun refreshFinished()
    fun setRefreshOverView(hiOverView: HiOverView?)

    /**
     * 设置下拉刷新的监听器
     * @param hiRefreshListener
     */
    fun setRefreshListener(hiRefreshListener: HiRefreshListener?)
    interface HiRefreshListener {
        fun onRefresh()
        fun enableRefresh(): Boolean
    }
}